File: archivist/src/test/java/com/zorroa/archivist/web/TaskControllerTests.java
package com.zorroa.archivist.web;

import com.zorroa.archivist.domain.*;
import com.zorroa.archivist.repository.TaskDao;
import com.zorroa.archivist.service.JobService;
import com.zorroa.sdk.zps.ZpsScript;
import org.junit.Before;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.mock.web.MockHttpSession;
import org.springframework.test.web.servlet.MvcResult;

import java.io.FileWriter;
import java.nio.file.Path;
import java.nio.file.Paths;

import static org.junit.Assert.assertEquals;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

/**
 * Created by chambers on 8/25/16.
 */
public class TaskControllerTests extends MockMvcTest {

    @Autowired
    JobService jobService;

    @Autowired
    TaskDao taskDao;

    JobSpec spec;
    Job job;
    TaskSpec tspec;
    Task task;

    @Before
    public void init() {
        JobSpec spec = new JobSpec();
        spec.setName("bar");
        spec.setType(PipelineType.Export);
        job = jobService.launch(spec);

        tspec = new TaskSpec();
        tspec.setName("start");
        tspec.setScript(new ZpsScript());
        tspec.setJobId(job.getJobId());
        task = jobService.createTask(tspec);
    }

    @Test
    public void testStreamLog() throws Exception {
        MockHttpSession session = admin();

        Path logFile = Paths.get(taskDao.getExecutableTask(task.getTaskId()).getLogPath());
        new FileWriter(logFile.toFile()).append("bilbo baggins").close();

        MvcResult result = mvc.perform(get("/api/v1/tasks/" + task.getTaskId() + "/_log")
                .session(session)
                .contentType(MediaType.TEXT_PLAIN_VALUE))
                .andExpect(status().isOk())
                .andReturn();

        String data = result.getResponse().getContentAsString();
        assertEquals("bilbo baggins", data);
    }
}

File: archivist/src/test/kotlin/com/zorroa/archivist/web/TaskControllerTests.kt
package com.zorroa.archivist.web

import com.zorroa.archivist.domain.Pipeline
import com.zorroa.archivist.domain.emptyZpsScript
import com.zorroa.archivist.service.JobService
import com.zorroa.common.domain.Job
import com.zorroa.common.domain.JobSpec
import com.zorroa.common.domain.Task
import com.zorroa.common.domain.TaskSpec
import org.junit.Before
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.test.context.web.WebAppConfiguration
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultMatchers
import kotlin.test.assertEquals

@WebAppConfiguration
class TaskControllerTests : MockMvcTest() {

    @Autowired
    lateinit var jobService: JobService

    lateinit var task: Task

    @Before
    fun init() {
        val job = launchJob()
        task = jobService.createTask(job, TaskSpec("bar", emptyZpsScript("bar")))
    }

    fun launchJob() : Job {
        val spec = JobSpec("test_job",
                emptyZpsScript("foo"),
                args=mutableMapOf("foo" to 1),
                env=mutableMapOf("foo" to "bar"))

        return jobService.create(spec)
    }

    @Test
    fun testGet() {
        val session = admin()
        val result = mvc.perform(MockMvcRequestBuilders.get("/api/v1/tasks/" + task.id)
                .session(session)
                .contentType(MediaType.APPLICATION_JSON_VALUE))
                .andExpect(MockMvcResultMatchers.status().isOk)
                .andReturn()

        val t1 = deserialize(result, Task::class.java)
        assertEquals(task.id, t1.id)
    }

}